src/rt/aaA.d

```diff
@@ -376,8 +396,10 @@ extern (C) void* _aaGetY(AA* aa, const TypeInfo_AssociativeArray ti, in size_t v
     immutable hash = calcHash(pkey, ti.key);
 
     // found a value => return it
-    if (auto p = aa.findSlotLookup(hash, pkey, ti.key))
+    if (auto p = aa.findSlotLookup(hash, pkey, ti.key)) {
+        found = true;
         return p.entry + aa.valoff;
+    }
 
     auto p = aa.findSlotInsert(hash);
     if (p.deleted)
```

Comment:
I think I should be able to avoid the `p.entry =  allocEntry...`  when doing getOrAdd, but not sure how yet.